src/processors/universal.ts

```diff
@@ -370,6 +372,182 @@ class UniversalProcessor extends BaseOrderProcessor {
     return unlockTx;
   }
 
+  private unlockOrder(
+    orderId: string,
+    order: OrderData,
+    context: OrderProcessorContext
+  ) {
+    const giveChain = order.give.chainId;
+    // filling batch queue
+    let orderIds = this.unlockBatchesOrderIdMap.get(giveChain);
+    if (!orderIds) {
+      orderIds = new Set();
+    }
+    orderIds.add(orderId);
+    this.unlockBatchesOrderIdMap.set(giveChain, orderIds);
+    this.ordersDataMap.set(orderId, order);
+
+    // check that process is blocked
+    if (this.isBatchUnlockLocked) {
+      this.context.logger.debug("batch unlock processing is locked");
+      return;
+    }
+
+    this.isBatchUnlockLocked = true; // lock batch unlock locker
+
+    // execute batch unlock processing for full batch
+    if (orderIds.size >= this.params.batchUnlockSize) {
+      this.performBatchUnlock(giveChain, context);
+    }
+  }
+
+  private async performBatchUnlock(
+    giveChain: ChainId,
+    context: OrderProcessorContext
+  ) {
+    const logger = this.context.logger.child({
+      func: "performBatchUnlock",
+      giveChain,
+    });
+    logger.info("Batch unlocking is started");
+
+    try {
+      const orderIds = Array.from(
+        this.unlockBatchesOrderIdMap.get(giveChain)!
+      ).slice(0, this.params.batchUnlockSize);
+
+      const { result, unlockedOrders } = await this.tryUnlockBatch(
+        giveChain,
+        orderIds,
+        context
+      );
+
+      // clean executed orders form queue
+      unlockedOrders.forEach((id) => {
+        this.unlockBatchesOrderIdMap.get(giveChain)!.delete(id);
+        this.ordersDataMap.delete(id);
+      });
+
+      // check a full of batch
+      if (result) {
+        for (const [
+          chainId,
+          orderIds,
+        ] of this.unlockBatchesOrderIdMap.entries()) {
+          if (orderIds.size >= this.params.batchUnlockSize) {
+            this.performBatchUnlock(chainId, context); // start unlocking for not full batch
+            return;
+          }
+        }
+      }
+
+      // unlock batch process if each chain is not full
+      this.isBatchUnlockLocked = false;
+      logger.debug("All batches is not full");
+    } catch (e) {
+      logger.error(`Batch unlocking is failed ${e}`);
+    }
+  }
+
+  async tryUnlockBatch(
```

Comment:
This method is still too complex. 100 LOCs wrapped with one try-catch is a bad-smelling practice.